app/src/views/CountryPreparedness/PrivateCountryPreparedness/index.tsx

```diff
@@ -629,9 +646,45 @@ function PrivateCountryPreparedness() {
                     heading={strings.componentRatingResultsHeading}
                     withHeaderBorder
                     childrenContainerClassName={styles.ratingResultsContent}
+                    footerClassName={styles.legendContent}
+                    footerContent={(
+                        <div className={styles.legend}>
+                            <Heading
+                                level={5}
+                            >
+                                {strings.typeOfOperation}
+                            </Heading>
+                            <div className={styles.separator} />
+                            {Object.entries(areaComponentColorShades).map(([areaNum, color]) => {
+                                const area = assessmentStats?.topRatedComponents.find(
+                                    (component) => component.area.area_num === Number(areaNum),
+                                );
+
+                                if (!area) {
+                                    return null;
+                                }
```

Comment:
Let's create a dictionary for top rated components instead of using `find` inside a `map`